Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,11 +186,42 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    bool isHighlights = false;
+
+    if (!markedTexts.isEmpty()) {
+        // Get rid of duplicates in vector, check if contains highlights
+        ListHashSet<MarkedText> noDuplicates;
+        for (auto& element : markedTexts) {
+            noDuplicates.add(element);
+            if (!isHighlights && element.type == MarkedText::Type::Highlight)
+                isHighlights = true;
+        }
+        // Add back to markedTexts
+        if (!noDuplicates.isEmpty()) {
+            markedTexts.clear();
+            markedTexts.appendRange(noDuplicates.begin(), noDuplicates.end());
```

Comment:
I guess I'm not sure what we're doing here, pushing the duplicated to the back of the vector?

Reply:
Before, there would be a lot of duplicate highlights with the same ranges, which would cause problems in this case (lower priority, higher priority, duplicate lower priority). Thus, to combat this I weed out the duplicate highlight and ranges to ensure we aren't going to repaint over the expected behavior. However, I'll check again if my latest changes affect this.

Reply:
I just checked and there are still duplicates. However, since I order them afterwards, having duplicates should not affect the painting order anymore so it should still work regardless of getting rid of duplicates or not.